range_queries/mo_algorithm.cpp

```diff
@@ -0,0 +1,108 @@
+/**
+ *
+ * @file
+ * \brief A C++ Program to compute unique elements of ranges for different range
+ * queries
+ *
+ * \details
+ * The preprocessing part takes O(m Log m) time.
+ * Processing all queries takes O(n * √n) + O(m * √n) = O((m+n) * √n) time.
```

Comment:
Added "Unknown" author as I don't really know who created the original PR. You should add yourself, too, as you have improved the code a lot.
```suggestion
 * Processing all queries takes O(n * √n) + O(m * √n) = O((m+n) * √n) time.
 *
 * \author Unknown author
 * \author [Abhinn Mishra](https://github.com/mishraabhinn)
```